modules/document_repository/jsx/categoryForm.js

```diff
@@ -0,0 +1,241 @@
+import PropTypes from 'prop-types';
+/**
+ * Document Upload Form
+ *
+ * Fetches data from Loris backend and displays a form allowing
+ * to upload a doc file
+ *
+ * @author Shen Wang
+ * @version 1.0.0
+ *
+ * */
+class DocCategoryForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      Data: {},
+      formData: {},
+      uploadResult: null,
+      errorMessage: null,
+      isLoaded: false,
+      loadedData: 0,
+      uploadProgress: -1,
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.isValidForm = this.isValidForm.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.uploadFile = this.uploadFile.bind(this);
+  }
+
+  componentDidMount() {
+    let self = this;
```

Comment:
I don't think the use of `self` is necessary here. This function can be written using `this`, although this can be done in a refactor.

Reply:
done

Reply:
It has to store the context into a variable as "this" reference will not be available in the callback.if  not  will cause an error "TypeError: Cannot read property 'setState' of undefined"

Reply:
Right, I guess I am used to using arrow functions in which case `this` is bound implicitly. This change can be done in a refactor.